src/parse_grammar.rs

```diff
@@ -1,16 +1,21 @@
 // HACK(eddyb) silence warnings from unused exports in the generated code.
 #![allow(unused)]
 
+// HACK(eddyb) needed for bootstrapping `parse_grammar`.
+mod gll {
+    pub(crate) use crate::runtime;
+}
+
 include!(concat!(env!("OUT_DIR"), "/parse_grammar.rs"));
 
-use scannerless::Pat as SPat;
+use crate::scannerless::Pat as SPat;
 use std::ops::Bound;
 use std::str::FromStr;
 
-impl<Pat: From<SPat>> FromStr for ::grammar::Grammar<Pat> {
-    type Err = ::runtime::ParseError<::runtime::LineColumnRange>;
+impl<Pat: From<SPat>> FromStr for crate::grammar::Grammar<Pat> {
```

Comment:
Can you import `grammar` and `runtime`? Same logic, no `crate` outside of imports.